pkg/apiserver/domain/service/workflow.go

```diff
@@ -588,15 +588,37 @@ func (w *workflowServiceImpl) TerminateRecord(ctx context.Context, appModel *mod
 	return nil
 }
 
+// ResumeWorkflow resume workflow
+func ResumeWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application) error {
+	app.Status.Workflow.Suspend = false
+	steps := app.Status.Workflow.Steps
+	for i, step := range steps {
+		if step.Type == wfTypes.WorkflowStepTypeSuspend && step.Phase == common.WorkflowStepPhaseRunning {
+			steps[i].Phase = common.WorkflowStepPhaseSucceeded
+		}
+		for j, sub := range step.SubStepsStatus {
+			if sub.Type == wfTypes.WorkflowStepTypeSuspend && sub.Phase == common.WorkflowStepPhaseRunning {
+				steps[i].SubStepsStatus[j].Phase = common.WorkflowStepPhaseSucceeded
+			}
+		}
+	}
+	if err := kubecli.Status().Patch(ctx, app, client.Merge); err != nil {
+		return err
+	}
+	return nil
+}
+
 // TerminateWorkflow terminate workflow
 func TerminateWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application) error {
 	// set the workflow terminated to true
 	app.Status.Workflow.Terminated = true
+	// set the workflow suspend to false
+	app.Status.Workflow.Suspend = false
 	steps := app.Status.Workflow.Steps
 	for i, step := range steps {
 		switch step.Phase {
 		case common.WorkflowStepPhaseFailed:
-			if step.Reason != custom.StatusReasonFailedAfterRetries {
+			if step.Reason != custom.StatusReasonFailedAfterRetries && step.Reason != custom.StatusReasonTimeout {
```

Comment:
why we need to add this check？

Reply:
When you try to terminate a workflow, if the step is already `failedAfterRetries` or `timeout`, then its reason shouldn't be `terminate` since its already finished.